test/puppetlabs/pcp/messaging_test.clj

```diff
@@ -119,6 +119,23 @@
             (is (= (:target message) (:target @received)))
             (is (= "pcp://client01.example.com/agent" (:sender @received)))))))))
 
+;; Test that the client allows specifying a sender and it won't be overwritten.
```

Comment:
I tried to get the following working, as it's a stronger assertion that the message is unchanged, but couldn't get the redef right.
```
(deftest sender-spoof-test
  (with-app-with-config app broker-services broker-config
    (let [message (-> (message/make-message)
                      (assoc :target "pcp://client01.example.com/agent"
                             :sender "pcp://client02.example.com/agent"
                             :message_type "example/any_schema"))
          received (promise)]
      (with-open [sender (connect-client "client01" (constantly true))]
        (client/wait-for-connection sender (* 40 1000))
        (with-redefs [gniazdo.core/send-msg (fn [ws msg]
                                              (println "Message" msg)
                                              (deliver received msg))]
          (client/send! sender message)
          (is (= message (message/decode (deref received 5000 nil)))))))))
```